packages/frontend/src/state/lp/hooks.ts

```diff
@@ -0,0 +1,115 @@
+import { nearestUsableTick, TickMath } from '@uniswap/v3-sdk'
+import { fromTokenAmount } from '@utils/calculations'
+import { useAtomValue } from 'jotai'
+import { addressesAtom, isWethToken0Atom } from '../positions/atoms'
+import BigNumber from 'bignumber.js'
+import { OSQUEETH_DECIMALS } from '@constants/index'
+import { controllerContractAtom, controllerHelperHelperContractAtom, nftManagerContractAtom, quoterContractAtom, squeethPoolContractAtom } from '../contracts/atoms'
+import useAppCallback from '@hooks/useAppCallback'
+import { addressAtom } from '../wallet/atoms'
+import { Contract } from 'web3-eth-contract'
+import { useHandleTransaction } from '../wallet/hooks'
+import { ethers } from 'ethers'
+import { useCallback } from 'react'
+import { useGetDebtAmount, useGetTwapSqueethPrice, useGetVault } from '../controller/hooks'
+
+/*** GETTERS ***/
+
+export const useGetPosition = () => {
+    const contract = useAtomValue(nftManagerContractAtom)
+  
+    const getPosition = useCallback(
+      async (uniTokenId: number) => {
+        if (!contract) return null
+        const position = await contract.methods.positions(uniTokenId).call()
+        const { nonce, operator, token0, token1, fee, tickLower, tickUpper, liquidity, feeGrowthInside0LastX128, feeGrowthInside1LastX128, tokensOwed0, tokensOwed1 } = position
+        return {
+          nonce,
```

Comment:
It'd be great if we could define getting position using the useQuery hook from react-query with the key of ['position,' uniTokenId] so that we can take advantage of cache and get loading status, error message, etc.

Reply:
@daryakaviani Are we going to useQuery as Adam suggested?

Reply:
I didn't add it since this function is just used in the other functions. Was thinking we can take advantage of the useQuery perks via the preexisting `useLPPositionsQuery` if we need to display user LP positions. Isn't this the case? @adam5909099 @KMKoushik 

If not, let me know if I need to make a different subgraph query to handle positions specifically generated by the API.